kotest-assertions/kotest-assertions-klock/src/commonMain/kotlin/io.kotest/matchers/date/date.kt

```diff
@@ -0,0 +1,136 @@
+package io.kotest.matchers.date
+
+import com.soywiz.klock.Date
+import com.soywiz.klock.DateTime
+import io.kotest.matchers.Matcher
+import io.kotest.matchers.MatcherResult
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
+
+infix fun Date.shouldHaveSameYearAs(date: Date) = this shouldBe haveSameYear(date)
+
+infix fun Date.shouldNotHaveSameYearAs(date: Date) = this shouldNotBe haveSameYear(date)
+
+fun haveSameYear(date: Date) = object : Matcher<Date> {
+   override fun test(value: Date) =
+      MatcherResult(
+         value.year == date.year,
+         { "$value should have year ${date.year}" },
+         { "$value should not have year ${date.year}" }
+      )
+}
+
+infix fun Date.shouldHaveYear(year: Int) = this shouldBe haveYear(year)
+
+infix fun Date.shouldNotHaveYear(year: Int) = this shouldNotBe haveYear(year)
+
+fun haveYear(year: Int) = object : Matcher<Date> {
+   override fun test(value: Date) =
+      MatcherResult(
+         value.year == year,
+         { "$value should have year $year" },
+         { "$value should not have year $year" }
+      )
+}
+
+infix fun Date.shouldHaveSameMonthAs(date: Date) = this shouldBe haveSameMonth(date)
+
+infix fun Date.shouldNotHaveSameMonthAs(date: Date) = this shouldNotBe haveSameMonth(date)
+
+fun haveSameMonth(date: Date) = object : Matcher<Date> {
+   override fun test(value: Date) =
+      MatcherResult(
+         value.month == date.month,
+         { "$value should have month ${date.month}" },
+         { "$value should not have month ${date.month}" }
+      )
+}
+
+infix fun Date.shouldHaveMonth(month: Int) = this shouldBe haveMonth(month)
+
+infix fun Date.shouldNotHaveMonth(month: Int) = this shouldNotBe haveMonth(month)
+
+fun haveMonth(month: Int) = object : Matcher<Date> {
+   override fun test(value: Date) =
+      MatcherResult(
+         value.month.index1 == month,
+         { "$value should have month $month" },
+         { "$value should not have month $month" }
+      )
+}
+
+infix fun Date.shouldHaveSameDayAs(date: Date) = this shouldBe haveSameDay(date)
+
+infix fun Date.shouldNotHaveSameDayAs(date: Date) = this shouldNotBe haveSameDay(date)
+
+fun haveSameDay(date: Date) = object : Matcher<Date> {
+   override fun test(value: Date) =
+      MatcherResult(
+         value.day == date.day,
+         { "$value should have day ${date.day}" },
+         { "$value should not have day ${date.day}" }
+      )
+}
+
+infix fun Date.shouldHaveDay(day: Int) = this shouldBe haveDay(day)
+
+infix fun Date.shouldNotHaveDay(day: Int) = this shouldNotBe haveDay(day)
+
+fun haveDay(day: Int) = object : Matcher<Date> {
+   override fun test(value: Date) =
+      MatcherResult(
+         value.day == day,
+         { "$value should have day $day" },
+         { "$value should not have day $day" }
+      )
+}
+
+fun Date.shouldBeToday() = this shouldBe beToday()
```

Comment:
Here ```Be``` is coming twice so ```this shouldBe beToday``` also sounding off.
```this should beToday```  would be better.

Reply:
done.